pkg/processor/eventsource/eventsource.go

```diff
@@ -89,12 +89,11 @@ func (aes *AbstractEventSource) AllocateWorkerAndSubmitEvent(event nuclio.Event,
 
 		return nil, errors.Wrap(err, "Failed to allocate worker"), nil
 	}
+	// release worker when we're done
```

Comment:
Why? 
1. We need the worker to be released ASAP so that it can handle requests the moment it's done with this one. I've yet to read a guarantee that defer is called _immediately_ after the function returns (maybe you know otherwise)
2. Performance penalty using defer

Reply:
> 1. We need the worker to be released ASAP so that it can handle requests the moment it's done with this one. I've yet to read a guarantee that defer is called immediately after the function returns (maybe you know otherwise)

Since if there's a panic in `SubmitEventToWorker` we won't release the worker

> 2. Performance penalty using defer

I don't see any other option to guarantee that the worker will be release.

Reply:
`handleSubmitPanic` releases the worker :)

Reply:
I see. Less explicit but works - will remove.